The fuzzing support is non-building

It is out of sync with the current API.

```
error[E0609]: no field `train_rdo` on type `api::config::EncoderConfig`
  --> /home/lu_zero/rav1e/src/fuzzing.rs:90:14
   |
90 |   config.enc.train_rdo = g.g();
   |              ^^^^^^^^^ unknown field
   |
   = note: available fields are: `width`, `height`, `time_base`, `bit_depth`, `chroma_sampling` ... and 23 others

error[E0061]: this function takes 16 parameters but 13 parameters were supplied
   --> /home/lu_zero/rav1e/src/fuzzing.rs:210:7
    |
210 |     dec.encode_decode(
    |         ^^^^^^^^^^^^^ expected 16 parameters
    |
   ::: /home/lu_zero/rav1e/src/test_encode_decode/mod.rs:67:3
    |
67  | /   fn encode_decode(
68  | |     &mut self, w: usize, h: usize, speed: usize, quantizer: usize,
69  | |     limit: usize, bit_depth: usize, chroma_sampling: ChromaSampling,
70  | |     min_keyint: u64, max_keyint: u64, switch_frame_interval: u64,
...   |
139 | |     }
140 | |   }
    | |___- defined here
```

--- a/.github/workflows/rav1e.yml
+++ b/.github/workflows/rav1e.yml
@@ -57,6 +57,7 @@ jobs:
          - cargo-c
          - check-no-default
          - check-extra-feats
+         - fuzz
         include:
          - conf: 1.36.0-tests
            toolchain: 1.36.0
@@ -76,6 +77,8 @@ jobs:
            toolchain: stable
          - conf: check-extra-feats
            toolchain: stable
+         - conf: fuzz
+           toolchain: nightly-2019-12-02
 
     env:
       RUST_BACKTRACE: 1
@@ -126,7 +129,7 @@ jobs:
     - name: Add dav1d
       if: >
         matrix.conf == '1.36.0-tests' || matrix.conf == 'dav1d-tests' ||
-        matrix.conf == 'grcov-coveralls'
+        matrix.conf == 'grcov-coveralls' || matrix.conf == 'fuzz'
       env:
         LINK: http://www.deb-multimedia.org/pool/main/d/dav1d-dmo
         DAV1D_VERSION: 0.6.0-dmo1
@@ -232,6 +235,14 @@ jobs:
       if: matrix.conf == 'cargo-c'
       run: |
         cargo cbuild
+    - name: Install cargo-fuzz
+      if: matrix.conf == 'fuzz'
+      run: |
+        cargo install cargo-fuzz
+    - name: Run cargo-fuzz
+      if: matrix.conf == 'fuzz'
+      run: |
+        cargo fuzz build
     - name: Run cargo clean
       if: matrix.conf == 'grcov-coveralls'
       uses: actions-rs/cargo@v1

--- a/src/fuzzing.rs
+++ b/src/fuzzing.rs
@@ -86,8 +86,6 @@ pub fn fuzz_construct_context(data: &[u8]) {
   config.enc.tiles = g.g();
   config.enc.rdo_lookahead_frames = g.g();
   config.enc.speed_settings = SpeedSettings::from_preset(g.g());
-  config.enc.show_psnr = g.g();
-  config.enc.train_rdo = g.g();
 
   debug!("config = {:#?}", config);
 
@@ -190,8 +188,11 @@ pub fn fuzz_encode_decode(data: &[u8]) {
   let limit = (g.g::<u8>() % 3) as usize + 1;
   let min_keyint = g.g::<u64>() % 4;
   let max_keyint = g.g::<u64>() % 4 + 1;
+  let switch_frame_interval = 0;
   let low_latency = g.g();
+  let error_resilient = false;
   let bitrate = g.g();
+  let still_picture = false;
 
   debug!(
     "w = {:#?}\n\
@@ -217,9 +218,12 @@ pub fn fuzz_encode_decode(data: &[u8]) {
     Default::default(),
     min_keyint,
     max_keyint,
+    switch_frame_interval,
     low_latency,
+    error_resilient,
     bitrate,
     1,
     1,
+    still_picture,
   );
 }

--- a/src/fuzzing.rs
+++ b/src/fuzzing.rs
@@ -181,10 +181,10 @@ pub fn fuzz_encode_decode(data: &[u8]) {
 
   let mut g = create_generator!(data);
 
-  let w = g.g::<u8>() as usize + 1;
-  let h = g.g::<u8>() as usize + 1;
+  let w = g.g::<u8>() as usize + 16;
+  let h = g.g::<u8>() as usize + 16;
   let speed = 10;
-  let q = g.g();
+  let q = g.g::<u8>() as usize;
   let limit = (g.g::<u8>() % 3) as usize + 1;
   let min_keyint = g.g::<u64>() % 4;
   let max_keyint = g.g::<u64>() % 4 + 1;

